inference-engine/thirdparty/clDNN/src/gpu/loop_gpu.cpp

```diff
@@ -0,0 +1,478 @@
+
+        return ret;
+    }
+
+    std::vector<memory_impl::ptr> find_cropped_mem(const primitive_id& internal_id, const loop_inst& instance) {
+        const auto& input_iteration_mem = instance.input_iteration_mem;
+        for (const auto& iter_info : input_iteration_mem) {
+            if (iter_info.to_id == internal_id) {
+                return iter_info.sliced_mem;
+            }
+        }
+        const auto& output_iteration_mem = instance.output_iteration_mem;
+        for (const auto& iter_info : output_iteration_mem) {
+            if (iter_info.from_id == internal_id) {
+                return iter_info.sliced_mem;
+            }
+        }
+        return {}; // not found
+    }
+
+    void preprocess_output_memory(loop_inst& instance) {
+        auto& engine = instance.get_network().get_engine();
+        auto& output_iteration_mem = instance.output_iteration_mem;
+        auto body_network = instance.get_body_network();
+        const auto& output_primitive_map = node.get_output_mappings();
+        output_iteration_mem.reserve(output_primitive_map.size());
+        for (size_t i = 0; i < output_primitive_map.size(); ++i) {
+            const auto& primitive_map = output_primitive_map.at(i);
+            const primitive_id& external_id = primitive_map.external_id;
+            const primitive_id& internal_id = primitive_map.internal_id;
+            if (primitive_map.axis < 0) {
+                memory_impl::ptr memory = get_outer_output_memory(instance, external_id);
+                set_output_data(*body_network, internal_id, *memory);
+            } else {
+                memory_impl::ptr to_mem = get_outer_output_memory(instance, external_id);
+                auto output_prim = body_network->get_primitive(internal_id);
+                layout cropped_layout = output_prim->output_memory().get_layout();
+
+                const int max_iteration = node.get_max_iteration();
+                std::vector<memory_impl::ptr> cropped_mems;
+                cropped_mems.reserve(max_iteration);
+                for (int i=0; i < max_iteration; ++i) {
+                    memory_impl::ptr croped_mem = engine.allocate_memory(cropped_layout, 0);
+                    cropped_mems.push_back(croped_mem);
+                }
+
+                const int linear_size = static_cast<int>(cropped_layout.get_linear_size());
+                const int stride = linear_size * primitive_map.stride;
+                const int start = primitive_map.start < 0? node.get_max_iteration() - 1: primitive_map.start;
+                const int offset = linear_size * start;
+                cldnn::loop_inst::sliced_memory_binding memory_binding_info(
+                    primitive_map.internal_id, primitive_map.external_id,
+                    to_mem, cropped_mems, linear_size, stride, offset);
+                memory_binding_info.from_prim = body_network->get_primitive(internal_id);
+                output_iteration_mem.push_back(memory_binding_info);
```

Comment:
How about changing this "output_iteration_mem" as "concatenated_outputs"  to express more the characteristics?